### PR TITLE
Log a warning if the exception in case add thread flow is users not found

### PR DESCRIPTION
**Pull request by wssheldon**
Users are automatically engaged in Slack Case threads frequently who did not exist in the workspace. We can log this as a warning and move on.

Tested this locally by creating a case via an ingested Signal:

```bash
...
WARNING:Unable to resolve Slack participant: The request to the Slack API failed. (url: https://www.slack.com/api/users.lookupByEmail)
The server responded with: {'ok': False, 'error': 'users_not_found'}:/Users/wshel/Projects/dispatch/src/dispatch/plugins/dispatch_slack/service.py:emails_to_user_ids:72
DEBUG:function.elapsed.time.dispatch.plugins.dispatch_slack.plugin.SlackConversationPlugin.add_to_thread: 0.2627900000079535:/Users/wshel/Projects/dispatch/src/dispatch/decorators.py:wrapper:139
```

You can see above we now log a `WARNING` when the user cannot be resolved in the `add_to_thread` code flow.

### CODE REVIEW

File: src/dispatch/plugins/dispatch_slack/service.py [hunk 1]
```diff
@@ -34,6 +34,51 @@ def resolve_user(client: WebClient, user_id: str) -> dict:
...
+        >>> print(user_ids)
+        ["U01ABCDE1", "U01ABCDE2"]
+    """
+    user_ids = []
+
+    for participant in set(participants):
+        try:
+            user_id = resolve_user(client, participant)["id"]
+        except SlackApiError as e:
+            msg = f"Unable to resolve Slack participant: {e}"
```
Comment (whitdog47) [inline src/dispatch/plugins/dispatch_slack/service.py:69]:
```suggestion
            msg = f"Unable to resolve Slack participant: {p}, with error: {e}"
```

Comment (whitdog47) [reply src/dispatch/plugins/dispatch_slack/service.py:69]: Would it be helpful to also log the participant that didn't get resolved?

**Review by whitdog47 — CHANGES_REQUESTED** [review]

File: src/dispatch/plugins/dispatch_slack/service.py (hunk 1 above)
Comment (wssheldon) [inline src/dispatch/plugins/dispatch_slack/service.py:69]:
```suggestion
            msg = f"Unable to resolve Slack participant {participant}: {e}"
```

Comment (wssheldon) [reply src/dispatch/plugins/dispatch_slack/service.py]: Done

File: src/dispatch/plugins/dispatch_slack/service.py [hunk 2]
```diff
@@ -34,6 +34,51 @@ def resolve_user(client: WebClient, user_id: str) -> dict:
...
+
+    for participant in set(participants):
+        try:
+            user_id = resolve_user(client, participant)["id"]
+        except SlackApiError as e:
+            msg = f"Unable to resolve Slack participant {participant}: {e}"
+
+            if e.response["error"] == SlackAPIErrorCode.USERS_NOT_FOUND:
+                log.warning(msg)
+                continue
```
Comment (wssheldon) [inline src/dispatch/plugins/dispatch_slack/service.py:73]: I think this is redundant actually

**Review by whitdog47 — APPROVED** [review]

File: src/dispatch/plugins/dispatch_slack/service.py (hunk 2 above)
Comment (wssheldon) [reply src/dispatch/plugins/dispatch_slack/service.py:73]: The `continue`

Comment (wssheldon) [reply src/dispatch/plugins/dispatch_slack/service.py:73]: It is necessary, actually.

File: src/dispatch/plugins/dispatch_slack/service.py [hunk 3]
```diff
@@ -34,6 +34,51 @@ def resolve_user(client: WebClient, user_id: str) -> dict:
...
+
+            log.exception(msg)
```
Comment (wssheldon) [inline src/dispatch/plugins/dispatch_slack/service.py:75]:
```suggestion
            log.exception(msg)
            continue
```

File: src/dispatch/plugins/dispatch_slack/service.py (hunk 2 above)
Comment (whitdog47) [inline src/dispatch/plugins/dispatch_slack/service.py:73]:
```suggestion
            if e.response["error"] == SlackAPIErrorCode.USERS_NOT_FOUND:
                log.warning(msg)
            else:
                log.exception(msg)
```

Comment (whitdog47) [reply src/dispatch/plugins/dispatch_slack/service.py:73]: Might be clearer in intent

**Pull request merged**